src/towncrier/test/test_format.py

```diff
@@ -179,6 +181,158 @@ def test_basic(self):
+                    },
+                ),
+                (
+                    "Names",
+                    {},
+                ),
+                (
+                    "Web",
+                    {
+                        ("3", "bugfix", 0): "Web fixed.",
+                        ("2", "bugfix", 0): "Multi-line bulleted\n- fix\n- here",
+                    },
+                ),
+            ]
+        )
+
+        definitions = OrderedDict(
+            [
+                ("feature", {"name": "Features", "showcontent": True}),
+                ("bugfix", {"name": "Bugfixes", "showcontent": True}),
+                ("misc", {"name": "Misc", "showcontent": False}),
+            ]
+        )
+
+        expected_output = """# MyProject 1.0 (never)
+
+### Features
+
+- Fun! (baz)
+- Foo added. (#2, #9, #72)
+- Multi-line
+  here (#3)
+- Stuff! (#4)
+
+### Misc
+
+- bar, #1, #9, #142
+
+
+## Names
+
+No significant changes.
+
+
+## Web
+
+### Bugfixes
+
+- Multi-line bulleted
+  - fix
+  - here
+
+  (#2)
+- Web fixed. (#3)
+"""
+
+        template = pkg_resources.resource_string(
+            "towncrier", "templates/default.md"
+        ).decode("utf8")
+
+        fragments = split_fragments(fragments, definitions)
+        output = render_fragments(
+            template,
+            None,
+            fragments,
+            definitions,
+            ["-", "~"],
+            wrap=True,
+            versiondata={"name": "MyProject", "version": "1.0", "date": "never"},
+        )
+        self.assertEqual(output, expected_output)
+
+        # Also test with custom issue format
+        expected_output = """# MyProject 1.0 (never)
+
+### Features
+
+- Fun! ([baz])
+- Foo added. ([2], [9], [72])
+- Multi-line
+  here ([3])
+- Stuff! ([4])
+
+[baz]: https://github.com/twisted/towncrier/issues/baz
+[2]: https://github.com/twisted/towncrier/issues/2
+[3]: https://github.com/twisted/towncrier/issues/3
+[4]: https://github.com/twisted/towncrier/issues/4
+[9]: https://github.com/twisted/towncrier/issues/9
+[72]: https://github.com/twisted/towncrier/issues/72
+
+### Misc
+
+- [bar], [1], [9], [142]
+
+[bar]: https://github.com/twisted/towncrier/issues/bar
+[1]: https://github.com/twisted/towncrier/issues/1
+[9]: https://github.com/twisted/towncrier/issues/9
+[142]: https://github.com/twisted/towncrier/issues/142
+
+
+## Names
+
+No significant changes.
+
+
+## Web
+
+### Bugfixes
+
+- Multi-line bulleted
+  - fix
+  - here
+
+  ([2])
+- Web fixed. ([3])
+
+[2]: https://github.com/twisted/towncrier/issues/2
+[3]: https://github.com/twisted/towncrier/issues/3
+"""
+
+        output = render_fragments(
+            template,
+            "[{issue}]: https://github.com/twisted/towncrier/issues/{issue}",
+            fragments,
+            definitions,
+            ["-", "~"],
+            wrap=True,
+            versiondata={"name": "MyProject", "version": "1.0", "date": "never"},
+        )
+        self.assertEqual(output, expected_output)
```

Comment:
let's separate act and assert.

```suggestion
        )

        self.assertEqual(output, expected_output)
```